src/deepmd_wrapper.F

```diff
@@ -0,0 +1,125 @@
+MODULE deepmd_wrapper
+
+   USE ISO_C_BINDING,                   ONLY: C_CHAR,&
+                                              C_DOUBLE,&
+                                              C_INT,&
+                                              C_LOC,&
+                                              C_NULL_CHAR,&
+                                              C_PTR
+#include "./base/base_uses.f90"
+
+   IMPLICIT NONE
+   PRIVATE
+   PUBLIC :: dp_deep_pot, dp_deep_pot_compute
+
+#if(__DEEPMD)
+   INTERFACE
+      FUNCTION dp_deep_pot_c(c_model) BIND(C, name="DP_NewDeepPot")
+         USE ISO_C_BINDING, ONLY: C_PTR, &
+                                  C_CHAR, &
+                                  C_DOUBLE, &
+                                  C_INT, &
+                                  C_NULL_CHAR, &
+                                  C_LOC
+      CHARACTER(LEN=1, KIND=C_CHAR)                      :: c_model(*)
+      TYPE(C_PTR)                                        :: dp_deep_pot_c
+
+      END FUNCTION
+
+      SUBROUTINE dp_deep_pot_compute_c(dp, natom, &
+                                       coord, atype, cell, &
+                                       energy, force, virial, &
+                                       atomic_energy, atomic_virial) BIND(C, name="DP_DeepPotCompute")
+         USE ISO_C_BINDING, ONLY: C_PTR, &
+                                  C_INT, &
+                                  C_CHAR, &
+                                  C_DOUBLE
+      TYPE(C_PTR), VALUE                                 :: dp
+      INTEGER(C_INT), VALUE                              :: natom
+      REAL(C_DOUBLE), DIMENSION(natom, 3), INTENT(IN)    :: coord
+      INTEGER(C_INT), DIMENSION(natom), INTENT(IN)       :: atype
+      REAL(C_DOUBLE), DIMENSION(9), INTENT(IN)           :: cell
+      REAL(C_DOUBLE), INTENT(OUT)                        :: energy
+      REAL(C_DOUBLE), DIMENSION(natom, 3), INTENT(OUT)   :: force
+      REAL(C_DOUBLE), DIMENSION(9), INTENT(OUT)          :: virial
+      REAL(C_DOUBLE), DIMENSION(natom), INTENT(OUT)      :: atomic_energy
+      REAL(C_DOUBLE), DIMENSION(natom, 9), INTENT(OUT)   :: atomic_virial
+
+      END SUBROUTINE
+   END INTERFACE
+#endif
+
+CONTAINS
+
+! **************************************************************************************************
+!> \brief ...
+!> \param model ...
+!> \return ...
+! **************************************************************************************************
+
+   FUNCTION dp_deep_pot(model) RESULT(pot)
+      CHARACTER(len=*), INTENT(IN), TARGET               :: model
+      TYPE(C_PTR)                                        :: pot
+
+      CHARACTER(len=1, kind=C_CHAR)                      :: c_model(LEN_TRIM(model) + 1)
+      INTEGER                                            :: i, N
+
+      N = LEN_TRIM(model)
+      DO i = 1, N
+         c_model(i) = model(i:i)
+      END DO
+      c_model(N + 1) = C_NULL_CHAR
+#if(__DEEPMD)
+      pot = dp_deep_pot_c(c_model)
```

Comment:
You will run into issues if this function is (accidentally) called if CP2K was built without DeepMD.